src/subdataframe/subdataframe.jl

```diff
@@ -112,12 +112,21 @@ index(sdf::SubDataFrame) = index(sdf.parent)
 nrow(sdf::SubDataFrame) = ncol(sdf) > 0 ? length(sdf.rows)::Int : 0
 ncol(sdf::SubDataFrame) = length(index(sdf))
 
+function Base.getindex(sdf::SubDataFrame, colind::Union{ColumnIndex, Colon})
+    return sdf.parent[sdf.rows, colind]
```

Comment:
It might just be me (and might be a hangover from Python) but `:` always meant "copy" to me, so a colon in either slot (row or column) would crystalise the row index and return a DataFrame.